code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -0,0 +1,298 @@
+
+GLOBAL_LIST_EMPTY(hive_datums) // init by makeDatumRefLists()
+
+/datum/hive_status
+	var/name = "Normal"
+	var/hivenumber = XENO_HIVE_NORMAL
+	var/mob/living/carbon/Xenomorph/Queen/living_xeno_queen
+	var/slashing_allowed = XENO_SLASHING_ALLOWED //This initial var allows the queen to turn on or off slashing. Slashing off means harm intent does much less damage.
+	var/queen_time = QUEEN_DEATH_TIMER //5 minutes between queen deaths
+	var/xeno_queen_timer
+	var/hive_orders = "" //What orders should the hive have
+	var/color = null
+	var/prefix = ""
+	var/list/xeno_leader_list
+	var/list/xenos_by_typepath
+	var/list/xenos_by_tier
+
+/datum/hive_status/New()
+	. = ..()
+	xeno_leader_list = list()
+	xenos_by_typepath = list()
+	xenos_by_tier = list()
+
+	for(var/mob/living/carbon/Xenomorph/X in subtypesof(/mob/living/carbon/Xenomorph))
+		xenos_by_typepath[initial(X.caste_base_type)] = list()
+
+	for(var/tier in GLOB.xenotiers)
+		xenos_by_tier[tier] = list()
+
+/datum/hive_status/proc/remove_leader(var/mob/living/carbon/Xenomorph/X)
+	xeno_leader_list -= X
+	X.queen_chosen_lead = FALSE
+
+/datum/hive_status/proc/add_leader(var/mob/living/carbon/Xenomorph/X)
+	xeno_leader_list += X
+	X.queen_chosen_lead = TRUE
+
+/datum/hive_status/proc/get_total_xeno_number()
+	. = 0
+	for(var/t in xenos_by_tier)
+		. += length(xenos_by_tier[t])
+
+// doing this by type means we get a pseudo sorted list
+/datum/hive_status/proc/get_watchable_xenos()
+	var/list/xenos = list()
+	for(var/typepath in xenos_by_typepath)
+		if(typepath == /mob/living/carbon/Xenomorph/Queen) // hardcoded check for now
+			continue
+		for(var/i in xenos_by_typepath[typepath])
+			var/mob/living/carbon/Xenomorph/X = i
+			if(is_centcom_level(X.z))
+				continue
+			xenos += X
+	return xenos
+
+/datum/hive_status/proc/update_leader_pheromones()
+	for(var/i in xeno_leader_list)
+		var/mob/living/carbon/Xenomorph/X = i
+		X.handle_xeno_leader_pheromones(living_xeno_queen)
+
+/datum/hive_status/proc/update_queen()
+	if(living_xeno_queen)
+		return TRUE
+	if(length(xenos_by_typepath[/mob/living/carbon/Xenomorph/Queen]))
+		for(var/i in xenos_by_typepath[/mob/living/carbon/Xenomorph/Queen])
+			var/mob/living/carbon/Xenomorph/Queen/Q = i
+			if(is_centcom_level(Q.z))
+				continue
+			living_xeno_queen = Q
+			xeno_message("<span class='xenoannounce'>A new Queen has risen to lead the Hive! Rejoice!</span>",3)
+			update_leader_pheromones()
+			return TRUE
+	xeno_message("<span class='xenoannounce'>A sudden tremor ripples through the hive... the Queen has been slain! Vengeance!</span>",3)
+	xeno_message("<span class='xenoannounce'>The slashing of hosts is now permitted.</span>",2)

```

Comment:
Should only happen if it wasn't before ?

Reply:
please explain?  it's meant to properly handle having multiple queens in a hive, if the 'queen' queen dies another one replaces it otherwise the death message displays

Reply:
I meant this should only display if slashing wasn't permitted before, but it was nitpicking : the playerbase is experienced enough to not be confused by this.